walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -143,6 +100,58 @@
       if (newFree.duration() >= duration) {
         freeTimes.add(newFree);
       }
+
+      // We skip over optional blocks while making modifications to reqFreeTimes
+      if (curr.isReq()) {
+        TimeRange newFreeReq = TimeRange.fromStartEnd(reqStart, thisStart, false);
+        if (newFreeReq.duration() >= duration) {
+          reqFreeTimes.add(newFreeReq);
+        }
+        reqStart = thisEnd;
+      } else {
+        /*
+         * Checks if this is a valid optional block with fewer busy attendees than the existing
+         * minimum (in which case it replaces the minimum with the current value) or equal busy
+         * attendees than the existing minimum in which case it is added to the list of such
+         * ranges
+         */
+
+        /*
+         * Use this if else statement to try to convert the current block into a large
+         * enough block for the meeting
+         */
+        if (curr.duration() >= duration) {
+          // This block is valid by itself
+
+          // In case this time block ends at the end of the day, make its end inclusive
+          if (thisEnd == TimeRange.END_OF_DAY) {
+            curr = TimeRange.fromStartEnd(thisStart, thisEnd, true);
+          }
+        } else {
+          // If this block doesn't work, we may be able to coalesce it with
+          // adjacent optional blocks and create a valid block
+
+          // Stores all optional attendees who are busy during the larger time block
+          HashSet<String> blockOptBusy = new HashSet<>();
+          int blockEnd = createCombinedBlock(curr, busyTimes, duration, blockOptBusy);
+
+          if (blockEnd != -1) {
+            if (blockEnd == TimeRange.END_OF_DAY) {
```

Comment:
In a nested if-else block, avoid making additional nested blocks if you can. For example, this could just be rewritten to: `curr = TimeRange.fromStartEnd(thisStart, blockEnd, blockEnd == TimeRange.END_OF_DAY);` to get rid of one nesting layer.